nebula/ui/components/VPNRecentConnections.qml

```diff
@@ -8,61 +8,18 @@ import QtQuick.Layouts 1.14
 import QtQml.Models 2.2
 
 import Mozilla.VPN 1.0
+import Mozilla.VPN.qmlcomponents 1.0
 
 ColumnLayout {
     property bool hasVisibleConnections: false
     property bool showMultiHopRecentConnections: true
     property real numVisibleConnections: recentConnectionsRepeater.count
-    property var recentConnectionsModel: getRecentConnectionsModel()
-
-    function getRecentConnectionsModel() {
-        const maxNumVisibleConnections = 2
-        const recentConnections = []
-        for (let i=1; i<VPNSettings.recentConnections.length; i++) {
-            if (recentConnections.length === maxNumVisibleConnections) {
-                return recentConnections;
-            }
-
-            const recentConnection = VPNSettings.recentConnections[i];
-            const servers = recentConnection.split(" -> ");
-            const isMultiHop = servers.length > 1;
-
-            if (isMultiHop !== showMultiHopRecentConnections) {
-                return recentConnections;
-            }
-
-            const connection = [];
 
-            for(let x = 0; x < servers.length; x++) {
-                let index = servers[x].lastIndexOf(",");
-                if (index <= 0) {
-                    console.log("Unable to parse server from " + servers[x]);
-                    continue;
-                }
-                let countryCode = servers[x].slice(index+1).trim();
-                let serverCityName = servers[x].slice(0, index).trim();
-
-                connection.push({
-                     countryCode: countryCode,
-                     serverCityName: serverCityName,
-                     localizedCityName: VPNLocalizer.localizedCityName(countryCode, serverCityName)
-                 });
-            }
-
-            const [{ localizedCityName: firstCityLocalizedName }, secondServer] = connection;
-            const accessibleLabel = secondServer
-                ? VPNl18n.MultiHopFeatureAccessibleNameRecentConnection
-                    .arg(firstCityLocalizedName)
-                    .arg(secondServer.localizedCityName)
-                : firstCityLocalizedName;
-
-            recentConnections.push({
-                isMultiHop,
-                connection,
-                accessibleLabel
-            });
-        }
-        return recentConnections
+    VPNFilterProxyModel {
+        property int count: 0
+        id: recentConnectionsModel
+        source: VPNRecentConnectionModel
+        filterCallback: (obj) => { return obj.isMultiHop === showMultiHopRecentConnections && ++count < 3; }
     }
```

Comment:
This seems like it will return the first 2 items in the recent connections model, but I think we actually want the 2nd and 3rd items, otherwise we will be showing the currently selected server as a recent connection